src/WPF/WPF.Viewer/Samples/Map/CreateDynamicBasemapGallery/CreateDynamicBasemapGallery.xaml.cs

```diff
@@ -0,0 +1,129 @@
+using System.Windows;
+using System.Windows.Media;
+using System.Windows.Controls;
+using System.Diagnostics;
+using System.Net.Http;
+
+namespace ArcGIS.WPF.Samples.CreateDynamicBasemapGallery
+{
+    [ArcGIS.Samples.Shared.Attributes.Sample(
+        name: "Create dynamic basemap gallery",
+        category: "Map",
+        description: "Implement a basemap gallery that automatically retrieves the latest customization options from the basemap styles service.",
+        instructions: "When launched, this sample displays a map containing a button that, when pressed, displays a gallery of all styles available in the basemap styles service. Selecting a style results in the drop-down menus at the base of the gallery becoming enabled or disabled. A disabled menu indicates that the customization cannot be applied to the selected style. Once a style and any desired customizations have been selected, pressing `Load` will update the basemap in the map view.",
+        tags: new[] { "basemap", "languages", "service", "style" })]
+    [ArcGIS.Samples.Shared.Attributes.OfflineData()]
+    public partial class CreateDynamicBasemapGallery
+    {
+        /// <summary>
+        /// A class to hold information about basemap styles, allowing for easy binding to UI elements and dynamic basemap creation.
+        /// </summary>
+        internal class BasemapStyleGalleryItem
+        {
+            public BasemapStyleInfo BasemapStyleInfo { get; set; }
+            public BasemapStyle BasemapStyle { get; set; }
+            public ImageSource ImageSource { get; set; }
+        }
+
+        public CreateDynamicBasemapGallery()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            // Create a new map with the ArcGIS Navigation basemap style.
+            MyMapView.Map = new Map(BasemapStyle.ArcGISNavigation);
+
+            // Create a new basemap styles service, pulling in the available styles and their information.
+            BasemapStylesServiceInfo service = await BasemapStylesServiceInfo.CreateAsync();
+
+            // Populate the basemap style gallery.
+            foreach (var styleInfo in service.StylesInfo)
+            {
+                // Remove all whitespace from the style name.
+                string styleName = styleInfo.StyleName.Replace(" ", string.Empty);
+
+                // Get an enum of type BasemapStyle using the formatted style name.
+                BasemapStyle style = (BasemapStyle)Enum.Parse(typeof(BasemapStyle), styleName, true);
+
+                try
+                {
+                    // Create a new basemap style gallery item.
+                    BasemapStyleGallery.Items.Add(new BasemapStyleGalleryItem
+                    {
+                        BasemapStyle = style,
+                        BasemapStyleInfo = styleInfo,
+                        ImageSource = await styleInfo.Thumbnail.ToImageSourceAsync()
```

Comment:
Currently this code executes synchronously (because we await ToImageSourceAsync loading all the items' bitmaps one by one) and so the progress bar spins for a while before users can see a gallery.

But, at least on WPF, you can bind ImageSource property to a Uri instead of a ready-to-use image, and WPF will gladly download it for you on-demand.  Just change this property to type "Uri" and set it to `styleInfo.Thumbnail.Source`, and you'll see the gallery list load instantly -- and thumbnail images  download as needed.

There's probably something similar possible on MAUI and WinUI, though I don't know it off the top of my head.